src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -443,21 +488,30 @@ private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExp
             }
             else
             {
+                // There's no start/length, so the length to use with a constructor is the element count.
                 lengthForConstructor = elementCount;
             }
 
-            if (!inPlace && !used)
+            if (inPlaceTarget is null && !used)
             {
-                // emitting a value that no one will see
+                // The caller has specified that we're creating a ReadOnlySpan expression that won't be used.
+                // We needn't emit anything.
                 return true;
             }
 
             if (elementCount == 0)
             {
-                if (inPlace)
+                // The span is empty.  If this is in-place initialization, call the default ctor.
```

Comment:
>// The span is empty.  If this is in-place initialization, call the default ctor. [](http://example.com/codeflow?start=16&length=81)

Is this code path covered by tests for the new element types that we now can handle?